### DISCUSSION TITLE
The recovered magnitude is different from that displayed

### DISCUSSION THREAD

**Discussion by Jocelyn1109** [Stellarium/stellarium#2487] [Q&A]
In Observing list when I add an object, for some objects, the recovered magnitude is different from that displayed.
(See picture below).
The magnitude is recovered in the code like this:
```
float objectMagnitude = selectedObject[0]->getVMagnitude(core);
QString objectMagnitudeStr = QString::number(objectMagnitude);
```
Is it normal ???

![magnitude](https://user-images.githubusercontent.com/43777557/173558498-734ae8c6-91ae-4f5d-a608-1b0812d61162.png)

**Comment by alex-w** [comment]
This is not normal. Where I can check the behavior?

**Reply by Jocelyn1109** [reply]
@alex-w Could you get my branch ? Is it possible for you ?

**Reply by alex-w** [reply]
DSO object may have 2 magnitudes in **B** and **V** photometric passbands. On the screen (infobox) you see magnitude in **B** photometric passband and in Observing list you see magnitude in **V** photometric passband (by the fact for _C60_ **V** magnitude is unknown = 99.). For DSO objects you need to use `getVMagnitude()` and, if **V** equals 99.0, then `getBMagnitude()` methods.

**Reply by Jocelyn1109** [reply]
@alex-w thanks, I will do that :)

**Comment by Jocelyn1109** [comment]
Hi @alex-w, in fact we can solve the problem for Nebula objects when magnitude = 99 because the method getBMagnitude exists only in Nebula.cpp. Unfortunately we cannot solve this problem for other types of objects like MeteorShower. The solution for the moment and for other types of objects is to leave the value of magnitude empty or to inform it with "unknown".
You could have a B Magnitude = 99 too (example with SH 2-246).

Here an example of what you can have after correction

![magnitude_1](https://user-images.githubusercontent.com/43777557/173809070-c01ac768-d0a2-4887-82b2-87155660f32d.png)
.

**Reply by alex-w** [reply]
Meteor showers hasn't magnitudes. In AstroCalc we use dash symbol for mark unknown magnitudes.

P.S. Magnitude 99. = unknown magnitude

**Comment by Jocelyn1109** [comment]
![magnitude_2](https://user-images.githubusercontent.com/43777557/173811306-958abc03-bbd2-47b4-a6ec-dfdb50dceff5.png)
